From: =?utf-8?b?SMOpY3RvciBPcsOzbiBNYXJ0w61uZXo=?= <zumbi@debian.org>
Date: Mon, 5 Mar 2018 01:24:07 +0100
Subject: dsc: keep everything
MIME-Version: 1.0
Content-Type: text/plain; charset="utf-8"
Content-Transfer-Encoding: base64

RGViaWFuIHBhY2thZ2UgYnVpbGQgZGVwZW5kZW5jaWVzIGFsbG93IGZvciBkaXJlY3Qgb3IgaW5k
aXJlY3QgYnVpbGQKY3ljbGVzLCB3aGVuIGRvaW5nIGEgZnVsbCBkaXN0cmlidXRpb24gYnVpbGQg
aW4gT0JTIHRoaXMgbWVhbnMga2VlcGluZwp0aGUga2VlcCBsaXN0IGdldHMgdmVyeSB0ZWRpb3Vz
IHZlcnkgcXVpY2tseS4KCkFkanVzdCB0aGUgZGVmYXVsdCBrZWVwIHByb2Nlc3Npbmcgc3VjaCB0
aGF0IGZvciBEZWJpYW4gKGRzYykgYnVpbGRzCmV2ZXJ5dGhpbmcgaXMga2VwdCBieSBkZWZhdWx0
CgpTaWduZWQtb2ZmLWJ5OiBTam9lcmQgU2ltb25zIDxzam9lcmQuc2ltb25zQGNvbGxhYm9yYS5j
by51az4KU2lnbmVkLW9mZi1ieTogSMOpY3RvciBPcsOzbiBNYXJ0w61uZXogPHp1bWJpQGRlYmlh
bi5vcmc+Cg==
---
 Build.pm | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/Build.pm b/Build.pm
index 59c17a5..8ae21b6 100644
--- a/Build.pm
+++ b/Build.pm
@@ -563,7 +563,9 @@ sub get_build {
     @directdepsend = grep {!/^-/} splice(@directdepsend, @deps + 1);
   }
   my @extra = (@{$config->{'required'}}, @{$config->{'support'}});
-  if (@{$config->{'keep'} || []}) {
+  if ($config->{'type'} eq 'dsc') {
+	  ;
+  } elsif (@{$config->{'keep'} || []}) {
     my %keep = map {$_ => 1} (@deps, @{$config->{'keep'} || []}, @{$config->{'preinstall'}});
     for (@{$subpacks || []}) {
       next if $keep{$_};
